### PR TITLE
Run iOS scenario apps on iPhone 11 and iOS 14

### PR DESCRIPTION
**Pull request by jmagman**
To facilitate upgrading to Xcode 13, run the tests on an iPhone 11 simulator, which is available on both Xcode 12 and 13. 

Succeeded: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8828886065297463201/+/u/test:_Scenario_App_Integration_Tests/stdout

Part of flutter/flutter#85555.
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m [hunk 1]
```diff
@@ -33,8 +33,14 @@ - (BOOL)compareGoldenToImage:(UIImage*)image {
     os_log_error(OS_LOG_DEFAULT, "GOLDEN DIFF FAILED: image does not exists.");
     return NO;
   }
-  CGImageRef imageRefA = [self.image CGImage];
-  CGImageRef imageRefB = [image CGImage];
+
+  // The home bar at the bottom of the screen can be different shades of gray
```
Comment (jmagman) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:37]: I was hoping to use [`prefersHomeIndicatorAutoHidden`](https://developer.apple.com/documentation/uikit/uiviewcontroller/2887510-prefershomeindicatorautohidden) similar to 
https://github.com/flutter/engine/blob/9e49ffb5859f722890015c2172d6bca233f70cd1/testing/scenario_app/ios/Scenarios/Scenarios/AppDelegate.m#L17
But that makes it worse, the home indicator bar animates in and then animates out.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m [hunk 2]
```diff
@@ -192,7 +192,7 @@ - (void)testPointerHover {
   XCTAssertNotEqual(lastHoverSequenceNumber, -1,
                     @"PointerChange.hover event did not occur for a hover");
   NSString* removeMessage = [NSString
-      stringWithFormat:@"%d,PointerChange.remove,device=0,buttons=0", lastHoverSequenceNumber + 1];
+      stringWithFormat:@"%ld,PointerChange.remove,device=0,buttons=0", lastHoverSequenceNumber + 1];
```
Comment (jmagman) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:195]: Fix a warning.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m (hunk 1 above)
Comment (cyanglaz) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:37]: This is ok to me.

Side node:
Is it possible to wait for the home indicator to be hidden with a `XCTestExpectation`?

**Review by cyanglaz — APPROVED** [review]
LGTM

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m [hunk 3]
```diff
@@ -50,6 +51,23 @@ - (instancetype)initWithLaunchArg:(NSString*)launchArg {
 }
 
 - (void)checkGoldenForTest:(XCTestCase*)test {
+  // Wait for the home bar indicator to disappear before comparing goldens.
+  XCUIApplication* springboard =
+      [[XCUIApplication alloc] initWithBundleIdentifier:@"com.apple.springboard"];
+  XCUIElement* homeGrabber = springboard.otherElements[@"Home Grabber"];
```
Comment (jmagman) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m:57]: > Is it possible to wait for the home indicator to be hidden with a `XCTestExpectation`?

Good idea! I checked springboard and there's a "Home Grabber" element.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/SpawnEngineTest.m [hunk 4]
```diff
@@ -2,37 +2,26 @@
...
-    [self addAttachment:screenshotAttachment];
-
-    XCTFail(@"Goldens do not match. Follow the steps in the "
-            @"README to update golden named %@ if needed.",
-            golden.goldenName);
-  }
+  XCUIElement* addTextField = application.textFields[@"ready"];
+  XCTAssertTrue([addTextField waitForExistenceWithTimeout:30]);
+
+  GoldenTestManager* manager =
```
Comment (jmagman) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/SpawnEngineTest.m:22]: Make this test use `GoldenTestManager` so it isn't doing one-off screenshots.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m (hunk 1 above)
Comment (jmagman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m]: > Is it possible to wait for the home indicator to be hidden with a `XCTestExpectation`?

Good idea! I checked springboard and there's a "Home Grabber" element.

**Comment by jmagman** [comment]
Grr, it failed and we don't upload the failing xc results bundle
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8828811169885771905/+/u/test:_Scenario_App_Integration_Tests/stdout

**Comment by jmagman** [comment]
> Grr, it failed and we don't upload the failing xc results bundle https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8828811169885771905/+/u/test:_Scenario_App_Integration_Tests/stdout

I'm not sure how to diagnose this in LUCI.  I'm going to revert to the unfortunate pixel cropping check, and revisit this change after upgrading to Xcode 13 on the bots.

**Review by cyanglaz — APPROVED** [review]
LGTM when CI passes

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m [hunk 5]
```diff
@@ -50,6 +51,23 @@ - (instancetype)initWithLaunchArg:(NSString*)launchArg {
...
+  NSPredicate* homeGrabberDisappear = [NSPredicate predicateWithFormat:@"exists == NO"];
+  XCTNSPredicateExpectation* homeGrabberExpectation =
+      [[XCTNSPredicateExpectation alloc] initWithPredicate:homeGrabberDisappear object:homeGrabber];
+  [test waitForExpectations:@[ homeGrabberExpectation ] timeout:10.0];
+
+  // Sometimes even when the home bar indicator element no longer exists
+  // there's still a phantom animation. Wait another second to let it finish animating.
+  XCTWaiterResult waitResult =
```
[the remark this answers is not in the document]
Comment (jmagman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenTestManager.m:65]: The animation in question is in springboard, not in our app.

**Pull request merged**